pandas/io/pytables.py

```diff
@@ -4334,34 +4349,51 @@ def read(
         stop: Optional[int] = None,
     ):
 
-        if not self.read_axes(where=where, start=start, stop=stop):
+        # validate the version
+        self.validate_version(where)
+
+        # infer the data kind
+        if not self.infer_axes():
             return None
 
+        result = self._read_axes(where=where, start=start, stop=stop)
+
         info = (
             self.info.get(self.non_index_axes[0][0], dict())
             if len(self.non_index_axes)
             else dict()
         )
-        index = self.index_axes[0].values
+
+        axes = list(self.axes)
+        inds = [i for i in range(len(axes)) if axes[i] is self.index_axes[0]]
+        assert len(inds) == 1
+        ind = inds[0]
+
+        index = result[ind][0]
+
         frames = []
-        for a in self.values_axes:
+        for i, a in enumerate(self.axes):
```

Comment:
confusing y ou are using self.axes here again

Reply:
we need `i` to do the lookup `result[i]` and `a` to do the check just below.  there may be a nicer way to do this if/when `values_axes` ceases to be stateful